src/data/literal.rs

```diff
@@ -233,36 +236,41 @@ impl<'a> Literal<'a> {
         match (self, other) {
             (Number(l), Number(r)) => {
                 if let (Ok(l), Ok(r)) = (l.parse::<i64>(), r.parse::<i64>()) {
-                    Ok(Number(Cow::Owned((l / r).to_string())))
+                    if r == 0 {
+                        Err(LiteralError::DivisorShouldNotBeZero.into())
+                    } else {
+                        Ok(Number(Cow::Owned((l / r).to_string())))
+                    }
                 } else if let (Ok(l), Ok(r)) = (l.parse::<f64>(), r.parse::<f64>()) {
-                    Ok(Number(Cow::Owned((l / r).to_string())))
-                } else {
-                    Err(LiteralError::UnreachableBinaryArithmetic.into())
-                }
-            }
-            (Number(l), Interval(r)) => {
-                if let Ok(l) = l.parse::<i64>() {
-                    Ok(Interval(l / *r))
-                } else if let Ok(l) = l.parse::<f64>() {
-                    Ok(Interval(l / *r))
+                    if r == 0.0 {
+                        Err(LiteralError::DivisorShouldNotBeZero.into())
+                    } else {
+                        Ok(Number(Cow::Owned((l / r).to_string())))
+                    }
                 } else {
                     Err(LiteralError::UnreachableBinaryArithmetic.into())
                 }
             }
             (Interval(l), Number(r)) => {
                 if let Ok(r) = r.parse::<i64>() {
-                    Ok(Interval(*l / r))
+                    if r == 0 {
+                        Err(LiteralError::DivisorShouldNotBeZero.into())
+                    } else {
+                        Ok(Interval(*l / r))
+                    }
                 } else if let Ok(r) = r.parse::<f64>() {
-                    Ok(Interval(*l / r))
+                    if r == 0.0 {
+                        Err(LiteralError::DivisorShouldNotBeZero.into())
+                    } else {
+                        Ok(Interval(*l / r))
+                    }
```

Comment:
Two `Err` in here are only the cases not covered by tests.
Let's add tests for generating those :)